libvast/fbs/partition_synopsis.fbs

```diff
@@ -13,6 +13,9 @@ table LegacyPartitionSynopsis {
 
   /// The import time range of this partition.
   import_time_range: vast.fbs.interval;
+
+  /// The version number of this partition.
```

Comment:
Can we expand a bit here? This is likely to be encountered in the future by someone without the context of the `rebuild` command, who will wonder why we have a separate version number in addition to the flatbuffer versioning itself.

Reply:
I added a bit of meat here, explaining why we need this.